### PR TITLE
networkmanager: Ensure that endianess is always set

### PR DESCRIPTION
**Pull request by martinpitt**
Initializing the endianess in parallel meant that a lot of IP addresses
in the UI got rendered with `byteorder === undefined`, and thus being in
reverse order on big-endian machines.

Make these errors apparent also on LE machines by asserting that the
value is set correctly. In the UI, initialize byte order first and defer
everything else until that finished, via a new "preinit" promise.

https://bugzilla.redhat.com/show_bug.cgi?id=1728213

### CODE REVIEW

**Comment by martinpitt** [comment]
Breaks the [firewall](https://209.132.184.41:8493/logs/pull-12382-20190722-091414-f9ae6335-cockpit-project-cockpit-verify-fedora-30/log.html#133) section, needs work.

**Comment by martinpitt** [comment]
Ah, this exposed a race condition which has been there before, but much smaller. This was possibly the reason of some test flakes before as well. Fixed in a separate commit.

File: pkg/networkmanager/interfaces.js [hunk 1]
```diff
@@ -207,6 +207,10 @@ function NetworkManagerModel() {
 
     var self = this;
 
+    /* resolved once first stage of initialization is done */
+    self._preinit_resolve = null;
+    self.preinit = new Promise((resolve, reject) => { self._preinit_resolve = resolve });
```
Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: What about moving the `client.call` from below to here?  That way we wouldn't have to introduce the `self._preinit_resolve` variable.

Like so:

```
self.preinit = new Promise((resolve, reject) => {
   client.call("/org/freedesktop/NetworkManager",
                "org.freedesktop.DBus.Properties", "Get",
                ["org.freedesktop.NetworkManager", "State"], { flags: "" })
            .fail(complain)
            .done(function(reply, options) {
                if (options.flags) {
                    if (options.flags.indexOf(">") !== -1)
                        utils.set_byteorder("be");
                    else if (options.flags.indexOf("<") !== -1)
                        utils.set_byteorder("le");
                    resolve();
                }
            });
});
```

Comment (martinpitt) [reply pkg/networkmanager/interfaces.js]: You are right, this is much more elegant. Fixed!

**Pull request merged**